core/unit_test/TestSharedSpace.cpp

```diff
@@ -104,9 +104,10 @@ TEST(defaultdevicetype, shared_space) {
                                Kokkos::DefaultHostExecutionSpace>)
     GTEST_SKIP() << "Skipping as host and device are the same space";
 
-#if defined(KOKKOS_ARCH_AMD_GPU) && !defined(KOKKOS_ARCH_AMD_GFX90A)
-  GTEST_SKIP()
-      << "skipping because specified arch does not support page migration";
+#if defined(KOKKOS_ARCH_AMD_GPU) && defined(KOKKOS_ENABLE_HIP)
+  if (!Kokkos::SharedSpace().impl_hip_driver_check_page_migration())
```

Comment:
Yeah, I sorta like SharedSpace only so that it catches... that exact case of "whoops, we called the impl function on something that wasn't `HIPManagedSpace`" at compile time, but it's better to do:

```
static_assert(std::is_same_v<Kokkos::SharedSpace, Kokkos::HIPManagedSpace>, "Unexpected shared space detected!");
if (!Kokkos::HIPManagedSpace().impl_hip_driver_check_page_migration())
```
?

Reply:
I'm fine either way; both versions fail in similar ways. No reason to bikeshed. 🙂